server/bleep/src/webserver/answer.rs

```diff
@@ -332,6 +345,103 @@ pub(super) async fn _handle(
     Ok(Sse::new(Box::pin(stream)))
 }
 
+#[derive(serde::Deserialize)]
+pub struct Explain {
+    pub relative_path: String,
+    pub line_start: usize,
+    pub line_end: usize,
+    pub branch: Option<String>,
+    pub repo_ref: RepoRef,
+    #[serde(default = "default_thread_id")]
+    pub thread_id: uuid::Uuid,
+}
```

Comment:
With this change we're handling the requests very differently across `/answer` and `/explain`. In `/answer` we're using the bloop query language to specify branch information, whereas here we're explicitly passing as a param.

We could simplify things by using the bloop query language for both types of request, and using the same `Params` struct that we used to. The client can build a query of the form: `Explain lines {} - {} in {} branch:<> path:<>` and pass as `q`. This saves us the need for the `virtual_req`.

Reply:
As discussed offline, we can't merge the two as `/answer/explain` needs to pre-fetch file data to synthesize a code chunk, and needs direct access to line numbers to construct a focused chunk.